src/EditorFeatures/CSharp/BlockCommentEditing/BlockCommentEditingCommandHandler.cs

```diff
@@ -170,7 +171,39 @@ private static bool IsCaretInsideBlockCommentSyntax(SnapshotPoint caretPosition)
             var trivia = syntaxTree.FindTriviaAndAdjustForEndOfFile(caretPosition, CancellationToken.None);
 
             var isBlockComment = trivia.IsKind(SyntaxKind.MultiLineCommentTrivia) || trivia.IsKind(SyntaxKind.MultiLineDocumentationCommentTrivia);
-            return isBlockComment && trivia.FullSpan.Start < caretPosition;
+            if (isBlockComment)
+            {
+                var span = trivia.FullSpan;
+                if (span.Start < caretPosition && caretPosition < span.End)
+                {
+                    return true;
+                }
+
+                // FindTriviaAndAdjustForEndOfFile always returns something if position is EOF,
+                // whether or not the result includes the position.
+                // And the SyntaxTrivia for block comments always ends on EOF, closed or not.
+                // So we need to handle
+                // /**/|EOF
+                // and
+                // /*  |EOF
+                if (caretPosition == snapshot.Length)
+                {
+                    if (span.Length < 4) // "/**/".Length
+                    {
+                        return true;
+                    }
+
+                    if (!trivia.ContainsDiagnostics)
```

Comment:
If the block comment is not closed, the SyntaxTrivia contains diagnostics. So if SyntaxTrivia is clean, the block comment should be closed. This if statement tries again to avoid string allocations below.